mountpoint-s3/src/main.rs

```diff
@@ -119,6 +119,14 @@ struct CliArgs {
     )]
     pub allow_other: bool,
 
+    #[clap(
+        long,
+        help = "Configure a string to be prepended to the 'User-Agent' HTTP request header for all S3 requests",
+        value_name = "PREFIX",
+        help_heading = CLIENT_OPTIONS_HEADER
+    )]
+    pub user_agent_prefix: Option<String>,
```

Comment:
I kind of think this should be an undocumented option. It's really only useful for us right now, right? Not sure it helps customers for it to show up in `--help`. But it would be our first undocumented option. Maybe an alternative is a new help section for "advanced options" or something, to make clear this isn't something most customers who aren't AWS would ever need to use.

Reply:
I'm actually less comfortable with moving it into a new "Advanced options", since someone looking for this kind of option would expect it to be with "Client options". Our intention is also for things like `--max-threads <>` and `--part-size <>` to be things most customers don't need to configure either.

At the moment, it shows up at the top of the list (since its based on the order it is defined). How do you feel about moving it to the bottom of the "Client options" list like below?

```
Client options:
      --maximum-throughput-gbps <N>  Maximum throughput in Gbps [default: auto-detected on EC2 instances, 10 Gbps elsewhere]
      --max-threads <N>              Maximum number of FUSE daemon threads [default: 16]
      --part-size <PART_SIZE>        Part size for multi-part GET and PUT [default: 8388608]
      --user-agent-prefix <PREFIX>   Configure a string to be prepended to the 'User-Agent' HTTP request header for all S3 requests
```

Reply:
Is anyone really going to be looking for it? Only AWS really gets to see the user agent. Unlike the other options, this is something a customer would basically never configure — it's really only useful for bundling Mountpoint into something else (and that basically means only useful for us).

Reply:
OK, let me move it to an "Advanced options" section at the bottom.